File: for-local-test/345.reverse-vowels-of-a-string.java
class Solution {
    public String reverseVowels(String s) {
        char[] chars = s.toCharArray();
        int left = 0;
        int right = chars.length - 1;
        while(left < right){
            while(left < right && !isVowel(chars[left])){
                left++;
            }
            while(left < right && !isVowel(chars[right])){
                right--;
            }
            if(left < right){
                char temp = chars[left];
                chars[left] = chars[right];
                chars[right] = temp;
                left++;
                right--;
            }
        }
        return new String(chars);
    }
    private boolean isVowel(char c){
        return "aeiouAEIOU".indexOf(c) >= 0;
    }

    public static void main(String[] args) {
        Solution solution = new Solution();
        String input = "hello";
        String output = solution.reverseVowels(input);
        System.out.println("Input: " + input);
        System.out.println("Output: " + output);
    }
}

File: for-local-test/python/345.reverse-vowels-of-a-string.py
class Solution:
    def reverseVowels(self, s: str) -> str:
        vowels = set("aeiouAEIOU")
        s_list = list(s)
        left, right = 0, len(s) - 1
        
        while left < right:
            while left < right and s_list[left] not in vowels:
                left += 1
            while left < right and s_list[right] not in vowels:
                right -= 1
            
            if left < right:
                s_list[left], s_list[right] = s_list[right], s_list[left]
                left += 1
                right -= 1
        
        return ''.join(s_list)

if __name__ == "__main__":
    solution = Solution()
    input_str = "hello"
    output_str = solution.reverseVowels(input_str)
    print(f"Input: {input_str}")
    print(f"Output: {output_str}")